### ISSUE TITLE
End date bug in tickstore _pandas_to_bucket

### ISSUE THREAD

**Issue by jamesblackburn** [manahl/arctic#15]
Looks like a potential issue - needs a test and 
https://github.com/isentium/arctic/commit/363cca98de77f7b40ab3c7d1fe5770752fb6a449

```
 @@ -531,7 +531,7 @@ def _ensure_supported_dtypes(self, array):
 
     def _pandas_to_bucket(self, df, symbol):
         start = self._to_dt(df.index[0].to_datetime())
-        end = self._to_dt(df.index[0].to_datetime())
+        end = self._to_dt(df.index[-1].to_datetime())
         rtn = {START: start, END: end, SYMBOL: symbol}
         rtn[VERSION] = CHUNK_VERSION_NUMBER
         rtn[COUNT] = len(df)
```

**Comment by cityhunterok** [comment]
yes, i read the source ,i'm puzzled, the bug is so clear, how you test it's read  write speed?

**Comment by jamesblackburn** [comment]
The answer is we don't use this code path - it was written as a demo.

We use `VersionStore` for minute bars and lower frequency (i.e. anything where we call `write` with a Pandas DataFrame).  For TickStore we have a Java writer, and, if we write in Python we use a `list` of `dict`s. The reason being that tick data is sparse (many columns, not all of them updating with each tick), and we want what's written to be sparse like the source data.

Perhaps try the `VersionStore`?

**Comment by jamesblackburn** [comment]
Fixed in 1.20.0
